realscale.py

```diff
@@ -37,10 +37,43 @@ def __init__(self):
         self.OptionParser.add_option("--unit", action="store", type="string", dest="unit", default="cm",
                                      help="Real-world unit")
 
+    def calc_scale_center(self, p1x, p1y, p2x, p2y):
+        """ Use straight line as scaling center.
+            - determine which point is center on basis of quadrant the line is in.
+            - approx this by using center of line
+            0,0 corresponds to UL corner of page
+        """
+        scale_center = [0,0] # resulting scaling point
+        # calc page center
+        pagecenter_x = self.unittouu(self.document.getroot().get('width'))/2
+        pagecenter_y = self.unittouu(self.document.getroot().get('height'))/2
+        # calc minmax of straightline ref points
+        minx = min(p1x,p2x)
+        maxx = max(p1x, p2x)
+        miny = min(p1y,p2y)
+        maxy = max(p1y, p2y)
+        # simplifiy calc by using center of line to determine quadrant
```

Comment:
My assumption is that this spot is at an edge of the drawing and therefore the drawing exists relative to this. But of course the line could be at any angle and in any part of the image. but the user is focussed on this part of the picture.
If they do the operation again - likely - then they won't have to scroll far.
Also, because the selection disappears :( - you can't do it live... :(